source/vir/src/modes.rs

```diff
@@ -32,7 +32,25 @@ pub fn mode_join(m1: Mode, m2: Mode) -> Mode {
     }
 }
 
-// placeholder for the erasure expected mode, which is computed after first mode checking pass
+// Placeholder for the erasure expected mode, which is computed after first mode checking pass.
+// (See in erase_expr(..., expect: Mode, ...) in erase.rs.)
+// We compute the erasure expected mode here so that the AIR/SMT generation can use it.
+// For example, if there are arithmetic operations that are exec (and therefore not erased),
+// then the AIR/SMT code will insert overflow checks for those arithmetic operations.
+// Example:
+//   fn test(#[exec] e: u64, #[spec] s: u64) {
+//     if e + 1 < s { ... }
+//   }
```

Comment:
I don't wanna hold up the PR with too much design discussion - but I wonder if we should require the user to be explicit in this case? e.g., if they want to create a `spec` if statement in the middle of `exec` code, they should have to write it like this:

```rust
#[proof] if e + 1 < s { ... }
```

Not just to make the mode-checking code easier, but also to keep user code clearer and more explicit

Reply:
Vale does it this way (you have to write "ghost if").  Dafny prefers to infer the ghostness.  I think we should have this discussion, but it's a broader discussion.  For example, it could be argued that _all_ proof code inside an exec function has to be marked as #[proof], even just calls to lemmas, to make the code easier for people to read and understand.  But then again, maybe not.  We should probably start a discussion page.